File: Arrays/twoNumberSum.js
// Find sum of two number and return indices
// https://leetcode.com/problems/two-sum/
const twoNumberSum1 = (nums, targetSum) => {
    const map = {}
    const numLen = nums.length
	for (let i = 0; i < numLen; i++) {
		const complement = targetSum - nums[i]
		if (complement in map) {
            return [map[complement], i]
        }
        map[nums[i]] = i
	}
}

// Find sum of two number and array of two numbers or empty array
//https://www.algoexpert.io/questions/Two%20Number%20Sum
const twoNumberSum2 = (array, target) => {
    const map = new Map(), 
				numLen = array.length
	for (let i = 0; i < numLen; i++) {
		const complement = targetSum - array[i]
		if (map.has(complement)) {
      return [complement, array[i]]
    }
    map.set(array[i], i)
	}
	return []
}

File: Arrays/two_number_sum.py
def twoNumberSum(array, targetSum):
    map = {}
    numLen = len(array)
    for i in range(numLen):
        complement = targetSum - array[i]
        if complement in map:
            return [complement, array[i]]
        map[array[i]] = i
    return []

def twoNumberSum2(nums, target):
    dict = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in dict:
            return [dict[complement], i]
        else:
            dict[num] = i